nfv_sriov.md

```diff
@@ -0,0 +1,253 @@
+
+```
+edpm_network_config_template
+  -
+    type: sriov_pf
+    name: nic3
+    numvfs: 10
+    use_dhcp: false
+    promisc: true
+```
+
+This example also assumes that the EDPM nodes:
+
+- PXE and boot settings are already configured.
+- Are at least one baremetal compute with SRIOV interfaces.
+
+Create an `OpenStackDataPlaneNodeSet` CR file,
+e.g. `dataplane_cr.yaml` to represent the EDPM nodes. See
+[dataplane_v1beta1_openstackdataplanenodeset.yaml](https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/samples/dataplane_v1beta1_openstackdataplanenodeset.yaml)
+for an example to modify as described in this document.
+
+Do not yet create the CR in OpenShift as the edits described in the
+next sections are required.
+
+The example
+[dataplane_v1beta1_openstackdataplanenodeset_sriov.yaml]https://github.com/openstack-k8s-operators/dataplane-operator/tree/main/examples/sriov)
+has SRIOV interfaces network configuration and required edpm ansible parameters.
+
+Modify your `OpenStackDataPlaneNodeSet` CR to set
+[edpm-ansible](https://github.com/openstack-k8s-operators/edpm-ansible)
+variables so that the
+[edpm_network_config role](https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/roles/edpm_network_config/)
+will configure networks with SRIOV interfaces.
+
+
+## Configure Nova
+
+Create a ConfigMap with content to be added to /etc/nova/nova.conf.d/,
+```yaml
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cpu-pinning-nova
+data:
+  25-cpu-pinning-nova.conf: |
+    [DEFAULT]
+    reserved_host_memory_mb = 4096
+    [compute]
+    cpu_shared_set = 0-3,24-27
+    cpu_dedicated_set = 8-23,32-47
+```
+The filename, e.g. `25-cpu-pinning-nova.conf`, must match `*nova*.conf` and
+the files are evaluated by Nova alphabetically (e.g. `01-foo-nova.conf`
+is processed before `02-bar-nova.conf`).
+
+```yaml
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: sriov-nova
+data:
+  03-sriov-nova.conf: |
+    [pci]
+    device_spec = {"vendor_id":"8086", "product_id":"1572", "address": "0000:19:00.3", "physical_network":"sriov-phy4", "trusted":"true"}
+```
+The filename, e.g. `03-sriov-nova.conf`, must match `*nova*.conf` and
+the files are evaluated by Nova alphabetically (e.g. `01-foo-nova.conf`
+is processed before `02-bar-nova.conf`).
+
+Create a custom version of the
+[nova service](https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/services/dataplane_v1beta1_openstackdataplaneservice_nova.yaml)
+which ships with the dataplane operator so that it uses the ConfigMap
+by adding it to the `configMaps` list.
+```yaml
+---
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneService
+metadata:
+  name: nova-custom-sriov
+spec:
+  label: dataplane-deployment-nova-custom-sriov
+  configMaps:
+    - cpu-pinning-nova
+    - sriov-nova
+  secrets:
+    - nova-cell1-compute-config
```

Comment:
These have a new format [1] for inclusion in `OpenStackDataPlaneService`.

[1] https://github.com/slagle/architecture/blob/ffbbba2cbd989dd18489c5bd43f83e0f344737c7/va/nfv/sriov/edpm/nodeset/nova_sriov.yaml#L22-L30

Reply:
On that note, there are several fields in that definition that relate to certs.  I'm not sure if you actually need the things under tlsCert -- I suspect not - but if you do, then that format has recently changed.  
https://github.com/openstack-k8s-operators/dataplane-operator/pull/875

On the other hand, you probably also need the caCerts field to mount and trust the caCerts so you can talk to other services.

Reply:
done